### ISSUE TITLE
Get power9 BMC issues resolved

### ISSUE THREAD

**Issue by naved001** [CCI-MOC/ops-issues#40]
The BMC seems to lose configuration when there's a power cut or the BMC is reset.

- [x] open ticket with IBM.
- [x] Send them the logs they asked for.
- [x] Do what they say after IBM looks at the logs.

**Comment by naved001** [comment]
IBM support didn't make it clear to me how to share the logs with them, and I can't see any open cases under my account on their website. So I emailed them the logs.

**Comment by msdisme** [comment]
@naved001 is the plan to also update firmware and see if it resolves?  (looking for more details because I want to start pointing the team from Israel at tickets like this-and eventually our backlog in general.

**Comment by naved001** [comment]
I think updating the firmware may fix it and I will go ahead and that today. Meanwhile, this is what IBM support wants me to do:

```

Perform a factory reset which may allow the BMC to boot from the  correct folder:                                                    

1 - collect command: cat /sys/class/watchdog/watchdog1/bootstatus  
2 - Factory reset                                                  
2 - Boot to petiboot and add config information                    
3 - run openbmctool to collect data                                
4 - Power down.                                                    
5 - Pull power                                                     
6 - Reconnect power and verify if IP is still there.              
7 - collect command: cat /sys/class/watchdog/watchdog1/bootstatus         
```

**Comment by naved001** [comment]
Updating the firmware did not fix the issue. I had to create the password again once the BMC rebooted.

**Comment by naved001** [comment]
bu-21-8 had an older firmware that I activated and that fixes the BMC issue. so that's one possible way of fixing it. The drawback is that the older firmware does not support `ipmitool`, and we have to use `openbmctool`.

**Comment by naved001** [comment]
> collect command: cat /sys/class/watchdog/watchdog1/bootstatus 

This command didn't exist, so I contacted IBM support, they then told me proceed with factory reset and I told them that I can't find such option. The closest one was `openbmctool`s `mc reset cold` which is just BMC reboot. So now they are trying to figure out what to do. I asked them for older firmware.

**Comment by naved001** [comment]
Still waiting for IBM support on this.

**Comment by naved001** [comment]
still blocked on IBM support.

**Comment by naved001** [comment]
IBM support got back to me with a detailed response on what to do and that fixed the BMC issues. This is what they told me to do:

> Sorry for the confusion in the previous action plan.
> 
> In order to run the bootstatus command you will need to ssh into the bmc to run the command, it is not an OS command as you experienced.
> 1. SSH into the BMC IP and run the following: cat /sys/class/watchdog/watchdog1/bootstatus  and collect output.
> 
> From the data that we are seeing it seems that after the power outage the BMC is not setting up the directories correctly and can not find the IP information when it boots. That is why we want to do a factory reset to the BMC, to see if we can recover the correct structure.
> 
> 2. Factory Reset command for this system is the following ipmitool command:
> 
> ipmitool -I lanplus -U <username> -P <password> -H <BMC_IP or Hostname> **raw 0x3A 0x11**
> 
> This will clear the bmc completely and all data will need to be re-entered and will get the BMC back to its original state.
> 
> 3. After system is recovered, go into the bmc and run the boot status command again:  cat /sys/class/watchdog/watchdog1/bootstatus 
> 
> 4. Pull power and see if IP stays and system boots normally.
> 
> I also saw your comment about other system in older level of code not having this issue. I would suggest to update these systems to the latest firmware that was just released, 940.20,  in order to apply some BMC and OPAL performance fixes to see if this stops the issue from reoccurring. If not then we will try to get the older code for you to install, or get you new BMC's.
> 
> If issue reoccurs collect new service data and send it with the bootstatus logs from before and after the factory reset and after the AC pull.

The important part is the raw command sent over ipmitool that resets the BMC.
